MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.m

```diff
@@ -334,12 +327,26 @@ - (NSString *)hideSecret:(NSString *)secret {
 
   // Hide everything if secret is shorter than the max number of displayed characters.
   NSUInteger appSecretHiddenPartLength =
-      (secret.length > kMSMaxCharactersDisplayedForAppSecret ? secret.length - kMSMaxCharactersDisplayedForAppSecret
-                                                             : secret.length);
+          (secret.length > kMSMaxCharactersDisplayedForAppSecret ? secret.length - kMSMaxCharactersDisplayedForAppSecret
+                  : secret.length);
   NSString *appSecretHiddenPart =
-      [@"" stringByPaddingToLength:appSecretHiddenPartLength withString:kMSHidingStringForAppSecret startingAtIndex:0];
+          [@"" stringByPaddingToLength:appSecretHiddenPartLength withString:kMSHidingStringForAppSecret startingAtIndex:0];
   return [secret stringByReplacingCharactersInRange:NSMakeRange(0, appSecretHiddenPart.length)
                                          withString:appSecretHiddenPart];
 }
 
+/**
+ * This is a method without implementation and is intended to be overwritten in all sub classes.
+ */
+- (NSArray *)retryIntervals {
```

Comment:
I would suggest something else: `retryIntervals` and `apiPath` being readonly properties and being set via a new initializer. That way we don't need the new `MSIngestionHttpSender` class and the empty implementations of `retryIntervals` and `apiPath`.